### ISSUE TITLE
Workflow of creating an order as an admin is broken

### ISSUE THREAD

**Issue by RachL** [openfoodfoundation/openfoodnetwork#4128]
## Description
When creating an order as an admin, pressing "continue" on the adjustments step will result in the loss of the order thus created.

The workaround is to not click on "continue", but go directly to the submenu "Payments". But for some users this workaround is difficult to find. We've had complaints in France.

## Steps to Reproduce
1. Log in as a manager (admin)
2.  Go to `/admin/orders/new`
3. (Select a shop if you are a super admin or a manager with several shop) and select an OC
4. Add products.
5. Save. This should get you to the next step: customer credential info
6. Add a customer.
7. Go to next step `/adjustments`
8. If you click on "continue" you are redirected to `/admin/orders` and your order is lost

## Workaround
9. Re-create your order again, but this time rather than clicking on "continue" go strait to "payments" in the right menu. As soon as a payment method is chosen your order will be correctly recorded.

## Animated Gif/Screenshot
![image](https://user-images.githubusercontent.com/6525576/62546566-fd276000-b863-11e9-80e2-943bf9d9af62.png)

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: v2.2.2
* Browser name and version: Firefox 68
* Operating System and version (desktop or mobile): Desktop

**Comment by luisramos0** [comment]
This is very bad and introduced with v2. We should fix this now right?
It will be easy to fix I think.

**Comment by RachL** [comment]
@luisramos0 is it more bad than the 5 bugs (as well introduced by v2) above this one?

It is an open question, I don't really have a strong opinion, just want to raise a flag in case there are things more urgent.

**Comment by luisramos0** [comment]
ah, ok, I just saw the new "top 5 bugs".
maybe we can take effort in consideration, let's see if those 5 are all achievable in a reasonable time frame. I think this one will be an easy one.

**Comment by myriamboure** [comment]
I see another issue : the "choose shipping method step" has disappeared, which I guess will break the system somehow as normally it is required to select a shipping method when you make an order... @RachL I saw that when updating French user guide.

**Comment by luisramos0** [comment]
I dont think thats a problem Myriam, if you dont select the shipping method
(below the line items), the order workflow will not move to completed.

**Comment by myriamboure** [comment]
Ok so I identified the problem by retesting:
1- I add a line item in the order. Below it is written that no shipping method can be chosen until a customer has been selected / entered.
2- When I click "update and recalculate fees" I'm taken to page 2, customer information, I fill them in or select a registered customer. If I click continue I'm taken to adjustment, and payment, etc.
3- SO I have to KNOW (without any indiciation of the system) that after I filled in the second tab "customer information" I have to go back to the previous tab "detail of the order" where I can know select a shipping method. By default, I realize the system has just taken the first one in the list, so it's not blocking the order to be made but there is a high risk of error of the hub manager making the order from back office not going back to first tab to select the good shipping method.

Is that clear @RachL @luisramos0 ? So for me there is a clear UX issue, it was better in v1 as there was a sub menu "delivery" just as there is a sub-menu "payment" for the order and the person filling in the order had to consciencely choose a shipping option for the user. For me it is a UX regression, and the feature is pretty much used.

**Comment by luisramos0** [comment]
Ah, yes. Makes sense Myriam, very bad ux.
I think this move of ship method to the first page came from spree, i
wonder how spree made this easier for the user...
Anyway, we can do whatever we think is best.
Maybe after customer details we make it jump back to ordrr details and flag
in some way that ship method needs to be filled? The adjustments tab is
there for the user to go there manually if she wants.

**Comment by kirstenalarsen** [comment]
ah ha!! this is the same issue as #4307, thanks @luisramos0 . Closing #4307. Flagging my agreement re. annoyance of users. Particularly the almost impossibility of finding how to set the shipping method! Also noting, that when we do fix this the user guide needs updating. Still reflects v1

**Comment by Matt-Yorkley** [comment]
I'm taking a look now. Looking at the steps to reproduce;

1. Log in as a manager (admin)
2. Go to /admin/orders/new
3. (Select a shop if you are a super admin or a manager with several shop) and select an OC

I also get a brief red error message after step 3 (before I have the chance to add products) that says "Line items can't be blank". This is unhelpful/confusing UX.

![Screenshot from 2019-12-13 13-19-13](https://user-images.githubusercontent.com/9029026/70799983-5f144b00-1dab-11ea-9582-c0928ee9026f.png)

**Comment by luisramos0** [comment]
yes, and a regression in v2. I have tried but I have not found an easy fix to it.

**Comment by luisramos0** [comment]
this error message must be shown if the last line item is removed from the order while editing it.

**Comment by Matt-Yorkley** [comment]
> this error message must be shown if the last line item is removed from the order while editing it.

Yeah, it needs to be there in most cases, but not this one.
